File: 310971/Stankevich/redis350/discussion/src/main/java/com/task/discussion/model/Notice.java
package com.task.discussion.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.cassandra.core.cql.Ordering;
import org.springframework.data.cassandra.core.cql.PrimaryKeyType;
import org.springframework.data.cassandra.core.mapping.Column;
import org.springframework.data.cassandra.core.mapping.PrimaryKeyColumn;
import org.springframework.data.cassandra.core.mapping.Table;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Table("tbl_notice")
public class Notice {

    @PrimaryKeyColumn(name = "country", ordinal = 0, type = PrimaryKeyType.PARTITIONED)
    private String country;

    @PrimaryKeyColumn(name = "tweetid", ordinal = 1, type = PrimaryKeyType.CLUSTERED, ordering = Ordering.DESCENDING)
    private Long tweetId;

    @PrimaryKeyColumn(name = "id", ordinal = 2, type = PrimaryKeyType.CLUSTERED, ordering = Ordering.DESCENDING)
    private Long id;

    @Column("content")
    private String content;

    @Column("state")
    private String state; // PENDING, APPROVE, DECLINE
}

File: 351003/brezgunov/discussion/src/main/kotlin/com/example/discussion/entity/Notice.kt
package com.example.discussion.entity

import org.springframework.data.cassandra.core.cql.PrimaryKeyType
import org.springframework.data.cassandra.core.mapping.PrimaryKeyColumn
import org.springframework.data.cassandra.core.mapping.Table
import java.util.UUID

@Table("tbl_notice")
class Notice(
    @PrimaryKeyColumn(type = PrimaryKeyType.PARTITIONED)
    var country: String,

    @PrimaryKeyColumn(type = PrimaryKeyType.CLUSTERED)
    var newsId: Long,

    @PrimaryKeyColumn(type = PrimaryKeyType.CLUSTERED)
    var id: UUID,

    var content: String,
)